ImageUploadToS3

shyam-dev10 opened the pull request.

shyam-dev10 committed:
updated

shyam-dev10 committed:
imageUploadtoS3

pavankpdev commented on package.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "dependencies": {
+    "aws-sdk": "^2.1574.0",
+    "fs": "^0.0.1-security"
```
fs is a default nodejs module, why install it again?

pavankpdev commented on package.json:
Why do you have package.json outside the project folders? Delete them

pavankpdev commented on package-lock.json:
Delete this

pavankpdev commented on server/package.json:
```diff
@@ -23,9 +24,10 @@
     "express": "^4.18.2",
     "express-async-handler": "^1.2.0",
     "express-validator": "^7.0.1",
+    "fs": "^0.0.1-security",
```
fs is a default nodejs module, why install it again?

pavankpdev commented on server/controller/product.js:
```diff
@@ -17,7 +18,16 @@ const createProduct = async (req, res) => {
 
-    const newProduct = new Product({ name, description, image, price, intervals, owner });
+    const file = req.files.image;
+    const imageUrl = await uploadFileToS3(file, 'services');
```
What is this `services` here?

pavankpdev commented on server/controller/product.js:
```diff
@@ -17,7 +18,16 @@ const createProduct = async (req, res) => {
+       price, 
+       intervals, 
+       owner });
     await newProduct.save();
```
Now need to do `.save()` here

pavankpdev reviewed (changes requested):
Is this route registered in the main.js file? I don't see that change. 

Also need the postman/Bruno test results

pavankpdev commented on server/services/services.js:
rename the file to fileUpload.js

shyam-dev10 committed:
Simplified deleteProduct controller code||renamed product Router names|| modified authm.js middleware removing the bearer string from the token and then verifying it

shyam-dev10 committed:
- Added multer middleware configuration to handle file uploads in the service code.
- Moved storage configuration to 'imageupload.js'.
- Updated 'uploadFileToS3' function to accept the file directly instead of the request object.
- Modified error handling in 'uploadFileToS3' to check if the file is undefined before accessing its properties.
- Removed unused imports and variables.

shyam-dev10 committed:
Update: modified update contoller function
Integrated AWS S3 for uploading product images, ensuring seamless storage and retrieval.
